File: editor/en/[165]Compare Version Numbers.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = list(map(int, version1.split(".")))
        v2 = list(map(int, version2.split(".")))
        length = max(len(v1), len(v2))

        for i in range(length):
            num1 = int(v1[i]) if i < len(v1) else 0
            num2 = int(v2[i]) if i < len(v2) else 0

            if num1 > num2:
                return 1
            elif num1 < num2:
                return -1

        return 0
# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[165]Compare Version Numbers.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function (version1, version2) {
    const v1 = version1.split('.');
    const v2 = version2.split('.');
    const length = Math.max(v1.length, v2.length);

    for (let i = 0; i < length; i++) {
        const num1 = i < v1.length ? parseInt(v1[i], 10) : 0;
        const num2 = i < v2.length ? parseInt(v2[i], 10) : 0;

        if (num1 !== num2) {
            return num1 > num2 ? 1 : -1;
        }
    }

    return 0;
};
//leetcode submit region end(Prohibit modification and deletion)
